src/r_segs.c

```diff
@@ -566,6 +567,42 @@ void R_StoreWallRange(const int start, const int stop)
   ds_p->curline = curline;
   rw_stopx = stop+1;
 
+  { // killough 1/6/98, 2/1/98: remove limit on openings
```

Comment:
It was in Boom and removed in MBF. Maybe we can just increase the `openings` array more and that will fix the crash too.

Reply:
Possible. But how can there be more openings than `video.width * video.height`? Do we have numbers how many openings we are missing?

Reply:
No idea, but this also fixes the crash:
`openings = Z_Calloc(1, video.width * video.height * 4 * sizeof(*openings), PU_RENDERER, NULL);`

Eternity has the same crash and they have MBF version:
```c
// killough 8/1/98: set static number of openings to be large enough
// (a static limit is okay in this case and avoids difficulties in r_segs.c)
#define MAXOPENINGS (MAX_SCREENWIDTH*MAX_SCREENHEIGHT)
short openings[MAXOPENINGS],*lastopening;
```

Reply:
Might still be a red herring. Something similar may happen if e.g. `start` and `stop` parameters are wrong.

Reply:
I copied this from PrBoom+, works since 98 I think: https://github.com/coelckers/prboom-plus/blob/969515162c5aebea4ad7a125ee178dcad3d576ad/prboom2/src/r_segs.c#L693-L723

Reply:
😆 Sure! However, could you do me a favor and track the values of `maxopenings` (and `pos` and `need`) in this loop in the map area that lead to the crash?

Reply:
It is happening on the wall of this room. Maybe it's not even acceptable? Something strange is happening:
![woof0003](https://github.com/user-attachments/assets/d6ecc3a5-b4a5-40fa-bbe9-0ce35af24d71)

Reply:
Does this result in a perceivable performance drop btw?

Reply:
> Does this result in a perceivable performance drop btw?

Not really.

Reply:
> I mean, seriously. Why allocate several megabytes of memory to keep track of more openings than we have pixels on the screen for a split second?

I like the patch, but I'm not sure if it's safe to just `return` from this function, could we break something? On the other hand, a few megs is nothing these days.

Reply:
> I like the patch, but I'm not sure if it's safe to just `return` from this function, could we break something? 

Should be safe, it's just rendering code. In a similar fashion, we return early from this function in non-overlay Automap mode.

> On the other hand, a few megs is nothing these days.

It's a matter of principle, a matter of "why should we?". To me, this falls into the same category as https://github.com/fabiangreffrath/woof/commit/14ed02e08dffc05549ac5c63e65ad17cd158020f

Reply:
Done in 107921c